src/main/java/com/learning/browsers/BrowserDriverGenerator.java

```diff
@@ -0,0 +1,30 @@
+package com.learning.browsers;
+
+import com.learning.configuration.PropertiesReader;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.edge.EdgeDriver;
+
+public class BrowserDriverGenerator implements BrowserDriver {
```

Comment:
I'd remove this interface... I don't see a point in creating interface for just one implementation.

Reply:
I think you could actually turn this class into `Browser` class and hide driver completely.
Then constructor would construct WebDriver object, store it in field and reuse.